api/src/main/java/io/cloudevents/Attributes.java

```diff
@@ -16,50 +16,48 @@
 package io.cloudevents;
 
 import java.net.URI;
+import java.time.ZonedDateTime;
 import java.util.Optional;
 
-import javax.validation.constraints.NotBlank;
-import javax.validation.constraints.NotNull;
-
-import com.fasterxml.jackson.annotation.JsonIgnore;
-
 /**
  * The marker interface for CloudEvents attributes
- *  
+ *
  * @author fabiojose
  *
  */
 public interface Attributes {
 
+    /**
+     * @return The version of the CloudEvents specification which the event uses
+     */
+    SpecVersion getSpecVersion();
+
 	/**
 	 * @return Identifies the event. Producers MUST ensure that source + id is unique for each distinct event
 	 */
-	@NotBlank
 	String getId();
 
 	/**
 	 * @return A value describing the type of event related to the originating occurrence.
 	 */
-	@NotBlank
 	String getType();
 
 	/**
 	 * @return The context in which an event happened.
 	 */
-	@NotNull
 	URI getSource();
 
 	/**
-	 * @return The version of the CloudEvents specification which the event uses
-	 */
-	@NotBlank
-	String getSpecversion();
-
-	/**
+     * TODO
 	 * A common way to get the media type of CloudEvents 'data';
 	 * @return If has a value, it MUST follows the <a href="https://tools.ietf.org/html/rfc2046">RFC2046</a>
 	 */
-	@JsonIgnore
-	Optional<String> getMediaType();
-	
+	Optional<String> getDataContentType();
+
+	Optional<URI> getDataSchema();
```

Comment:
Just my 2c:
`Optional` is a nice API, but it is costly (allocation wise). I wish we could sacrifice a bit of UX here in favour of performance and use `@Nullable`.

Having it as `Optional` would make it impossible to avoid allocations in third party performance critical implementations

Reply:
I'm trying to build 2 level APIs here like i did in sdk-go and sdk-rust:

* Event API: safer, more costly
* Message API: optimized for transferring events

Reply:
where `@Nullable` annotation comes from?

Reply:
it should come from `io.cloudevents` or a sub-package and be a JSR-305 meta-annotation, similar to:
https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/lang/Nullable.html

Reply:
> I'm trying to build 2 level APIs here

I just submitted an issue to discuss before we implement it:
https://github.com/cloudevents/sdk-java/issues/119

Reply:
Opened this one to discuss about it https://github.com/cloudevents/sdk-java/issues/123